### ISSUE TITLE
Broken seek to time function, on HLS feed.. jellyfin HLS controller creates Invalid M3U8 files

### ISSUE THREAD

**Issue by davesargrad** [jellyfin/jellyfin#5722]
Hi. I have jellyfin running.

I used [these ](https://github.com/home-cluster/jellyfin-openshift)instructions to get Jellyfin running on a fresh k8s 1.20 cluster.

![image](https://user-images.githubusercontent.com/48793689/113912020-ab597a80-97a8-11eb-9634-7519a11e57f3.png)

 I am able to play a 1650x1022 video back using the following [HLS URL](https://api.jellyfin.org/#operation/GetVariantHlsVideoPlaylist):
http://chad.corp.sensis.com:32004/Videos/75510f1b6c3d24954abec73b2329806c/main.m3u8?api_key=c6bf7f6eca6d458ebaed741a677e5699

I find that the playback is a bit sluggish. I also find that "seekto a point in time" fails  to _seek, or succeeds, but only after several minutes. 

One tell-tale issue I see is in the Jellyfin server log:
![image](https://user-images.githubusercontent.com/48793689/114028650-b6fb7e80-9846-11eb-89a0-160d9c60b9f4.png)

I see this happen whenever I click on the function that should seek to a point in time.
`[12:43:53] [WRN] [429] Jellyfin.Api.Controllers.DynamicHlsController: cannot serve /config/transcodes/7bfcae043c5fb93c677b4fcd2b82600554.ts as it doesn't exist and no transcode is running`

A quick search of the Jellyfin issue database uncovered [this](https://github.com/jellyfin/jellyfin/issues/2126). Is ffmpeg not part of the Jellyfin image that I picked up from docker? Shouldnt it be? 

I currently use [ReactPlayer ](https://www.npmjs.com/package/react-player)to playback my Jellyfin HLS stream. This seems to work ok for the  basic video playback function on the HLS stream that Jellyfin serves. 

However I am not able to reliably seek to a given point in time. 
I am able to perform functions such as "seek" just fine on other movies that I HLS stream from a Jellyfin instance  outside of kubernetes. I also have no problem streaming with this player from a variety of video sources, including HLS sources.

I need help isolating this extremely sluggish response to functions such as "seek to a point in time"

![image](https://user-images.githubusercontent.com/48793689/113917528-4f462480-97af-11eb-963f-9b3e703fed9c.png)

Question 1:
Why does the "zoom to a point in time" fail? 

Question 2: 
What logs can I view to see how Jellyfin is responding to commands that should govern which portion of the HLS stream are being streamed?

Question 3:
Given that the transcode is not found, and given issues such as [this](https://github.com/jellyfin/jellyfin/issues/2126), is ffmpeg not part of the Jellyfin image that I picked up from docker? Shouldnt it be? 

**Server System (please complete the following information):**
 - OS: [e.g. Docker on Kubernetes (1.20) on Linux]
 - Jellyfin Version: [e.g. 10.0.1]
![image](https://user-images.githubusercontent.com/48793689/113911133-9b8d6680-97a7-11eb-8fc9-4b69f732f5f3.png)
 - K8S Hardware Cluster  (8 core - 10 nodes)
 - model name      : Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz

 
**Client System (please complete the following information):**
Chrome: 
![image](https://user-images.githubusercontent.com/48793689/113911252-c5468d80-97a7-11eb-93e4-39f51e348341.png)

**Comment by davesargrad** [comment]
I just realized.. that the reason that "seek" likely doesnt work, is because [React Player ](https://www.npmjs.com/package/react-player) doesnt support Jellyfin.

I just posted an [issue ](https://github.com/cookpete/react-player/issues/1209)on the React Player issue database to see what I can learn from there. I didnt think of it but it may be that a code modification is required to implement the seek function.

**Comment by davesargrad** [comment]
I just verified that playback of the Jellyfin HLS stream works fine with ReactPlayer. My application works just fine as it streams HLS feeds from a Jellyfin instance that I stood up on a VirtualBox machine, rather than on kubernetes. Today I will see if I can isolate the problem. I'm still hoping to get thoughts from any of the Jellyfin experts, as to why "seek" and "fast-forward" functions might fail for streams I am receiving from my Jellyfin instance running inside kubernetes.

**Comment by davesargrad** [comment]
I just viewed the log from the jellyfin runtime service.. Every time I click on the "seek" function I see the following:
`[12:43:53] [WRN] [429] Jellyfin.Api.Controllers.DynamicHlsController: cannot serve /config/transcodes/7bfcae043c5fb93c677b4fcd2b82600554.ts as it doesn't exist and no transcode is running`

**Comment by davesargrad** [comment]
![image](https://user-images.githubusercontent.com/48793689/114030466-77ce2d00-9848-11eb-933d-7cbe891241dc.png)

**Comment by davesargrad** [comment]
Am I using the wrong docker image: 

image: docker.io/jellyfin/jellyfin

**Comment by davesargrad** [comment]
```
root@jellyfin-9fb5b756d-7987k:/# echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
root@jellyfin-9fb5b756d-7987k:/# ls /usr/local/bin
ocloc
root@jellyfin-9fb5b756d-7987k:/# ls /usr/sbin
addgroup   delgroup           groupadd     ldattach             pwck          tarcat                  validlocale
add-shell  deluser            groupdel     locale-gen           pwconv        tzconfig                vigr
adduser    dpkg-preconfigure  groupmems    mklost+found         pwunconv      update-ca-certificates  vipw
chgpasswd  dpkg-reconfigure   groupmod     newusers             readprofile   update-locale           zic
chmem      e2freefrag         grpck        nologin              remove-shell  update-passwd
chpasswd   e4crypt            grpconv      pam-auth-update      rmt           update-rc.d
chroot     e4defrag           grpunconv    pam_getenv           rmt-tar       useradd
cpgr       fdformat           iconvconfig  pam_timestamp_check  rtcwake       userdel
cppw       filefrag           invoke-rc.d  policy-rc.d          service       usermod
root@jellyfin-9fb5b756d-7987k:/# ls /usr/local/sbin
root@jellyfin-9fb5b756d-7987k:/# ls /usr/bin
'['                       delpart                   ipcrm              passwd             split
 addpart                  diff                      ipcs               paste              stat
 apt                      diff3                     ischroot           pathchk            stdbuf
 apt-cache                dircolors                 join               perl               sum
 apt-cdrom                dirname                   last               perl5.28.1         tabs
 apt-config               dpkg                      lastb              pinky              tac
 apt-get                  dpkg-deb                  lastlog            pldd               tail
 apt-key                  dpkg-divert               lcf                pr                 taskset
 apt-mark                 dpkg-maintscript-helper   ldd                printenv           tee
 arch                     dpkg-query                link               printf             test
 awk                      dpkg-split                linux32            prlimit            tic
 b2sum                    dpkg-statoverride         linux64            ptx                timeout
 base32                   dpkg-trigger              locale             realpath           toe
 base64                   du                        localedef          rename.ul          touch
 basename                 env                       logger             renice             tput
 bashbug                  expand                    logname            reset              tr
 captoinfo                expiry                    lsattr             resizepart         truncate
 catchsegv                expr                      lscpu              rev                tset
 chage                    factor                    lsipc              rgrep              tsort
 chattr                   faillog                   lslocks            runcon             tty
 chcon                    fallocate                 lslogins           savelog            tzselect
 chfn                     fincore                   lsmem              script             ucf
 choom                    find                      lsns               scriptreplay       ucfq
 chrt                     flock                     mawk               sdiff              ucfr
 chsh                     fmt                       mcookie            select-editor      unexpand
 cksum                    fold                      md5sum             sensible-browser   uniq
 clear                    getconf                   md5sum.textutils   sensible-editor    unlink
 clear_console            getent                    mesg               sensible-pager     unshare
 cmp                      getopt                    mkfifo             seq                update-alternatives
 comm                     gpasswd                   namei              setarch            users
 c_rehash                 gpgv                      nawk               setpriv            utmpdump
 csplit                   groups                    newgrp             setsid             wall
 cut                      head                      nice               setterm            wc
 debconf                  hostid                    nl                 sg                 whereis
 debconf-apt-progress     i386                      nohup              sha1sum            which
 debconf-communicate      iconv                     nproc              sha224sum          who
 debconf-copydb           id                        nsenter            sha256sum          whoami
 debconf-escape           infocmp                   numfmt             sha384sum          x86_64
 debconf-set-selections   infotocap                 od                 sha512sum          xargs
 debconf-show             install                   openssl            shred              yes
 deb-systemd-helper       ionice                    pager              shuf               zdump
 deb-systemd-invoke       ipcmk                     partx              sort
root@jellyfin-9fb5b756d-7987k:/# ls /sbin
agetty      ctrlaltdel   fdisk        fsfreeze       ldconfig     mkfs.ext3         raw                swapoff
badblocks   debugfs      findfs       fstab-decode   logsave      mkfs.ext4         resize2fs          swapon
blkdiscard  dumpe2fs     fsck         fstrim         losetup      mkfs.minix        runuser            switch_root
blkid       e2fsck       fsck.cramfs  getty          mke2fs       mkhomedir_helper  sfdisk             tune2fs
blkzone     e2image      fsck.ext2    hwclock        mkfs         mkswap            shadowconfig       unix_chkpwd
blockdev    e2label      fsck.ext3    installkernel  mkfs.bfs     pam_tally         start-stop-daemon  unix_update
cfdisk      e2mmpstatus  fsck.ext4    isosize        mkfs.cramfs  pam_tally2        sulogin            wipefs
chcpu       e2undo       fsck.minix   killall5       mkfs.ext2    pivot_root        swaplabel          zramctl
root@jellyfin-9fb5b756d-7987k:/# ls /bin
bash   date           echo     gzexe     mkdir       nisdomainname  run-parts  tar         vdir          zegrep
cat    dd             egrep    gzip      mknod       pidof          sed        tempfile    wdctl         zfgrep
chgrp  df             false    hostname  mktemp      pwd            sh         touch       which         zforce
chmod  dir            fgrep    ln        more        rbash          sleep      true        ypdomainname  zgrep
chown  dmesg          findmnt  login     mount       readlink       stty       umount      zcat          zless
cp     dnsdomainname  grep     ls        mountpoint  rm             su         uname       zcmp          zmore
dash   domainname     gunzip   lsblk     mv          rmdir          sync       uncompress  zdiff         znew

```

**Comment by davesargrad** [comment]
The linuxserver docker image seemed to fix my "seek" problem.

`LinuxServer.io image: linuxserver/jellyfin `

![image](https://user-images.githubusercontent.com/48793689/114038143-238b7300-9871-11eb-8f82-6a8e0b22de79.png)

I still see a huge lag on the delay function (10's of seconds). How do I isolate and refine why it takes so long to seek to a given time in my video?

Notice I now see this message:

![image](https://user-images.githubusercontent.com/48793689/114038363-559cd500-9871-11eb-8aa0-b8fa6ca1f27c.png)

**Comment by davesargrad** [comment]
From the time I clicked on seek, it took about 15 seconds before I saw this in the jellyfin log

![image](https://user-images.githubusercontent.com/48793689/114038889-c5ab5b00-9871-11eb-825f-05aed81daae4.png)


Then I waited for a while, but the video did not actually start playing again at that time.. so something still lurks.

**Comment by davesargrad** [comment]
The video I am trying to play back is about 13 minutes long. I just noticed that when I get into this failure mode, I can seek the start of the video. I can also seek to about 8 or 9 minutes in. In those two cases the video starts to play immediately at those times. Its when I seek to times between around 2 and 7 minutes, that the video does not start to play. Its almost as if there is a series of TS files missing.

**Comment by davesargrad** [comment]
I find the network timing, and gaps seemingly interesting.. The play actually continues during those times where there seems to be gaps.

![image](https://user-images.githubusercontent.com/48793689/114055869-21201d80-985f-11eb-905f-b8ad019424bd.png)

**Comment by davesargrad** [comment]
Here is some more data. I deleted my jellyfin deployment.. i reinstantiated it. I cleared the browser cache.

I registered the same media folder and image. I streamed it to the browser.. from time 0.

Within just a few seconds I found 64 ts files and 1 m3u8 file in the transcodes folder. 

![image](https://user-images.githubusercontent.com/48793689/114060714-2f813000-9885-11eb-81bc-5e6993899781.png)

I monitored the entire sequence of ts requests/responses in the browser diagnostic tools. There were no gaps. All of the 64 TS files were requested and downloaded in a timely fashion. Each TS files was roughly 12 to 23 seconds in duration.

**Comment by davesargrad** [comment]
This is the part I really find interesting. 

The m3u8 file that is currently in the jellyfin config/data/transcodes folder has 34 TS entries.

The initial had 64 entries. The sum of those two is 98. Indeed there are currently 98 TS files in that folder 
![image](https://user-images.githubusercontent.com/48793689/114062502-9989dd00-9865-11eb-9e2c-4406c79b6923.png)

I guess my question is.. is the m3u8 file getting "corrupted" somehow? It seems to change at the end of the played file.

**Comment by davesargrad** [comment]
I think I am close. I dont know why the m3u8 file is overwritten. I am doing a very controlled experiment to see that this happens again and again. 


1] Why is the m3u8 updated?
2] Why does it contain fewer TS entries?

See [here](https://github.com/cookpete/react-player/issues/1209#issuecomment-815975957).

**Comment by davesargrad** [comment]
Here is the log for the updated m3u8 file

![image](https://user-images.githubusercontent.com/48793689/114067091-6d248f80-986a-11eb-8d7b-c39e034061c4.png)

Notice that it occurs about 16 minutes after the first log was generated.

[root@qatar log]# cat FFmpeg.Remux-2021-04-08_16-56-26__2a5ec543.log
/Videos/75510f1b6c3d24954abec73b2329806c/hls1/main/64.ts

{"Protocol":0,"Id":"75510f1b6c3d24954abec73b2329806c","Path":"/data/media/videos/Hamburg/North/HamburgNorth.mp4","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"mov,mp4,m4a,3gp,3g2,mj2","Size":436275259,"Name":"HamburgNorth","IsRemote":false,"ETag":"543b6ca4c9f21c87d81daf7a932499c0","RunTimeTicks":8138162000,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":0,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"h264","CodecTag":"avc1","Language":"und","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/10000","CodecTimeBase":"1/20","Title":null,"VideoRange":"SDR","localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"1080p H264 SDR","NalLengthSize":"0","IsInterlaced":false,"IsAVC":false,"ChannelLayout":null,"BitRate":4284669,"BitDepth":8,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":true,"IsForced":false,"Height":1022,"Width":1650,"AverageFrameRate":10,"RealFrameRate":10,"Profile":"Constrained Baseline","Type":1,"AspectRatio":"825:511","Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p","Level":40,"IsAnamorphic":null},{"Codec":"aac","CodecTag":"mp4a","Language":"und","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/44100","CodecTimeBase":"1/44100","Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"Und - AAC - Mono - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"mono","BitRate":1378,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":1,"SampleRate":44100,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":"LC","Type":0,"AspectRatio":null,"Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":4288685,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/usr/lib/jellyfin-ffmpeg/ffmpeg -ss 00:06:24.000 -fflags +genpts -noaccurate_seek -f mov,mp4,m4a,3gp,3g2,mj2 -i file:"/data/media/videos/Hamburg/North/HamburgNorth.mp4" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -start_at_zero -vsync -1 -codec:a:0 copy -strict -2 -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 6 -hls_segment_type mpegts -start_number 64 -hls_segment_filename "/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b826005%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b826005.m3u8"


ffmpeg version 4.3.1-Jellyfin Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-gpl --enable-version3 --enable-static --enable-libfontconfig --enable-fontconfig --enable-gmp --enable-gnutls --enable-libass --enable-libbluray --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --arch=amd64 --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-vdpau --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvenc --enable-nvdec --enable-ffnvcodec
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'file:/data/media/videos/Hamburg/North/HamburgNorth.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp41isom
    creation_time   : 2021-03-19T11:51:09.000000Z
  Duration: 00:13:33.82, start: 0.000000, bitrate: 4288 kb/s
    Stream #0:0(und): Video: h264 (Constrained Baseline) (avc1 / 0x31637661), yuv420p, 1650x1022 [SAR 1:1 DAR 825:511], 4284 kb/s, 10 fps, 10 tbr, 10k tbn, 20 tbc (default)
    Metadata:
      creation_time   : 2021-03-19T11:51:09.000000Z
      handler_name    : VideoHandler
      encoder         : AVC Coding
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 1 kb/s (default)
    Metadata:
      creation_time   : 2021-03-19T11:51:09.000000Z
      handler_name    : SoundHandler
Output #0, hls, to '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b826005.m3u8':
  Metadata:
    encoder         : Lavf58.45.100
    Stream #0:0: Video: h264 (Constrained Baseline) (avc1 / 0x31637661), yuv420p, 1650x1022 [SAR 1:1 DAR 825:511], q=2-31, 4284 kb/s, 10 fps, 10 tbr, 90k tbn, 10 tbc (default)
    Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 1 kb/s (default)
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600564.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600565.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600566.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600567.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600568.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600569.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600570.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600571.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600572.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600573.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600574.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600575.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600576.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600577.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600578.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600579.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600580.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600581.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600582.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600583.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600584.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600585.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600586.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600587.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600588.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600589.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600590.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600591.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600592.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600593.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600594.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600595.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600596.ts' for writing
[hls @ 0x55a0f5d5d600] Opening '/config/data/transcodes/7bfcae043c5fb93c677b4fcd2b82600597.ts' for writing
frame= 4298 fps=0.0 q=-1.0 Lsize=N/A time=00:13:33.80 bitrate=N/A speed=2.51e+03x
video:224807kB audio:72kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown

**Comment by davesargrad** [comment]
Ok. The jellyfin server actually responds to a request for an m3u8 file with the following file
![image](https://user-images.githubusercontent.com/48793689/114074921-74609300-9894-11eb-978e-15d777ebac6c.png)

This has many more TS entries (135.ts is the last entry I see)
![image](https://user-images.githubusercontent.com/48793689/114075081-a540c800-9894-11eb-80ef-ae95fdd2a7d5.png)

So it seems as if the total number of TS files in the transcodes folder (98) is short. It also seems that the m3u8 file that is there, only addresses a portion of the data.

**Comment by davesargrad** [comment]
To me this feels like a bug in the HLS controller.

**Comment by cvium** [comment]
There's a lot going on here and I don't have time to go through it all but I can give you some pointers:

1. I hope your instance isn't accessible from the internet because you've posted your apikey multiple times.
2. The playlist generated by ffmpeg is not used by anything.
3. We generate our own VOD playlist that isn't technically valid because the EXTINF values are incorrect, which is because the segments aren't actually generated yet. Segments are generated as quickly as your hardware allows.
4. Seeking causes a new ffmpeg job to be spawned. If you seek to a segment that is already created it might just give you that instead of restarting the transcode.

**Comment by davesargrad** [comment]
By the way, I tried to reduce the image size and the frame rate by adding the following to the URL &width=825&height=511&framerate=4.0... 
This did not seem to have an effect. I'm not sure those parameters are actually used.

I chose the width and height to preserve the aspect ratio. Should these parameters have had an impact on the streamed HLS frames?

**Comment by davesargrad** [comment]
I'm trying to set up an experiment using curl.. I have successfully been able to curl the TS files that are listed in the playlist that the client receives. I then diff these to the actual TS files that are in the transcode folder, and I'm able to see that these are the same.

In the following 0.ts is a file I downloaded using curl. I ran diff against the 0'th file in the playlist, and they were the same.

I may be able to set up an experiment where I emulate the client using a command such as curl. I could wrap this in a python script to build out a client emulator. This would allow me to potentially isolate the problem further.

The experiment would have me request the intial playlist, then request each of the segments in order, or out of order. I could then compare the received files to the actual segment files.

It would probably take me a day or two to build this test client.

**Comment by davesargrad** [comment]
> 3\. We generate our own VOD playlist that isn't technically valid because the EXTINF values are incorrect, which is because the segments aren't actually generated yet. Segments are generated as quickly as your hardware allows.

Hi @cvium 

Could you please help me to understand this from the client applications perspective. Perhaps this is the core of the issue I am having. If the playlist is not technically valid, can I be sure that the number of TS files that it includes is correct?

**Comment by davesargrad** [comment]
Ok.. I've added callbacks to onBuffer, onBufferEnd, onStart, onEnd, onError, onSeek. Currently I simply print these events when they occur.

After I seek I see onBuffer but no onBufferEnd


![image](https://user-images.githubusercontent.com/48793689/114189033-bed72380-9917-11eb-9925-0e5ea036a23c.png)

I saw the 52.ts requested and downloaded upon "seek", so it would seem that the buffer should have ended.

![image](https://user-images.githubusercontent.com/48793689/114189169-e3cb9680-9917-11eb-9b1c-7448c95d0ca9.png)

**Comment by davesargrad** [comment]
@cvium 
I've now tried two different players ([ReactPlayer](https://www.npmjs.com/package/react-player and [VideoJS](https://videojs.com/)) with the HLS stream provided by Jellyfin.  They both can play  the content to some extent. However they are both flaky when I begin to use their ability to "seek" positions within the stream.

It strikes me that this may relate to the fact that the playlist is not valid as you mentioned.

At this point I will find a different HLS service. I think I've ruled out the client-side as a source of problem. Its too bad. I found Jellyfin quite attractive at first glance. I just dont think its HLS support is there yet.

I'll post my findings back here once I've integrated an alternative HLS service.

**Comment by davesargrad** [comment]
@cvium 
Ok. For the HLS service piece of my architecture, I've now replaced jellyfin with nginx in combination with an offline generation of the M3U8 and TS files (which I can easily script).

My service now works without the flakiness that I saw when using the Jellyfin HLS interface.

At this stage of the game, the issue (in my mind) is clearly the mismatch between the playlist created by the Jellyfin HLS controller, and the specification for HLS. 

Until this is fixed, the HLS controller within Jellyfin seems to be unusable as a general purpose HLS service.

**Comment by davesargrad** [comment]
It does strike me that Jellyfin could easily solve this problem, by using the M3U8 created by ffmpeg. FFmpeg could be invoked in the background as content is added, and the static files (m3u8 and ts) could be stored. This is a feature that you could disable by default, but enable, when the user configures HLS enabled.

**Comment by cvium** [comment]
You should read the spec.

The reason for the "invalid" playlist is that in order to seek _forward_ the playlist has to be complete ie. a VOD playlist. Our playlist is not valid because the `EXTINF` duration is _supposed_ to be the duration of the individual segments, **BUT** we do not know the lengths of the segments beforehand because the transcode happens on-the-fly.

**Comment by davesargrad** [comment]
@cvium 

No need to read the spec.. will not use jellyfin.. Ty though.
